Imports.md

```diff
@@ -0,0 +1,407 @@
+# ![wgsl-import-spec](branding.svg)
+
+# Summary
+
+We propose adding an importing mechanism to the WGSL shading language as an extension.
+
+# Motivation
+
+When writing bigger WGSL shaders, one tends to **split the code into reusable pieces**. Examples are re-using a set of lighting calculation functions and sharing a struct between two compute shaders and.
+
+However, current WGSL tooling is not built for this. The official language purposefully does not include this feature, nor does it provide adjacent features like namespaces, and there is no standardized extension yet.
+
+We also should account for **importing shader from libraries**. Ideally, users could upload WGSL shaders to existing package managers, which other users could then consume.
+
+Finally, we want **multiple tools** which can compile WGSL-with-imports down to raw WGSL. Using WGSL-with-imports both in Rust projects and in web projects should be possible.
+
+## Guide-level explanation
+
+The `import` statement extension is designed to appear somewhat familiar to TypeScript syntax,
+but with a Rust like recursive grammar to make importing
+multiple importable items less verbose.
+The syntax is also heavily inspired by [Gleam](https://gleam.run/).
+
+By placing an import at the very top of a file, one can either import an entire module, or only specific importable items, such as functions, structs or types.
+
+```
+// Importing a single item using a relative path
+import ./lighting/pbr;
```

Comment:
In contrast, this feels like I'm importing a module and could call its members using `pbr.<module_member_name>`. 

Also using a dot to refer to an item inside a module seems like it might introduce ambiguity and additional challenges with the rest of wgsl?

Reply:
I looked through the parsing and implemented it. It's unambiguous. 🎉 

I'm happy with requiring curly braces for importing items though. Maybe that actually makes them easier to read.

Reply:
Requiring curly braces at the end seems nice for people learning the syntax. 
perhaps start with that (and could still relax the syntax in the future.)

Reply:
adding this issue to TBD section